src/functions-and-arrays.js

```diff
@@ -1,73 +1,597 @@
+  }
+
+  for (let i = 0; i < array.length; i++) {
+    if (longestWord.length < array[i].length) {
+      longestWord = array[i];
+    } else {
+      continue;
+    }
+  }
+  return longestWord;
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(array) {
+  let sum = 0;
+
+  for (let i = 0; i < array.length; i++) {
+    if (typeof array[i] != "number") {
+      continue;
+    } else {
+      sum = sum + array[i];
+    }
+  }
+  return sum;
+}
+
+function sum(array) {
+  let sum = 0;
+
+  for (let i = 0; i < array.length; i++) {
+    if (
+      !(
+        typeof array[i] === "number" ||
+        typeof array[i] === "string" ||
+        typeof array[i] === "boolean"
+      )
+    ) {
+      throw new Error("Unsupported data type sir or ma'am");
+    } else if (typeof array[i] === "number") {
+      sum = sum + array[i];
+    } else if (typeof array[i] === "string") {
+      sum = sum + array[i].length;
+    } else if (typeof array[i] === "boolean") {
+      if (array[i] === true) {
+        sum++;
+      } else {
+        continue;
+      }
+    }
+  }
+  return sum;
+}
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(array) {
+  if (!array.length) {
+    return null;
+  }
+  let sum = sumNumbers(array);
+  let average = sum / array.length;
+  return average;
+}
+
 // Level 2: Array of strings
-const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+const wordsArr = [
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace",
+];
+
+function averageWordLength(array) {
+  if (!array.length) {
+    return null;
+  } else {
+    let sum = 0;
+    for (let i = 0; i < array.length; i++) {
+      sum = sum + array[i].length;
+    }
+    let average = sum / array.length;
+    return average;
+  }
+}
+
+function avg(arr) {
+  if (!arr.length) {
+    return null;
+  }
+  let total = sum(arr);
+
+  let average = total / arr.length;
+  return parseFloat(average.toFixed(2));
+}
 
 // Iteration #5: Unique arrays
 const wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+  "crab",
+  "poison",
+  "contagious",
+  "simple",
+  "bring",
+  "sharp",
+  "playground",
+  "poison",
+  "communion",
+  "simple",
+  "bring",
 ];
 
+function uniquifyArray(array) {
+  if (!array.length) {
+    return null;
+  } else {
+    array = Array.from(new Set(array));
+  }
+
+  return array;
+}
+
 // Iteration #6: Find elements
-const wordsFind = ['machine', 'subset', 'trouble', 'starting', 'matter', 'eating', 'truth', 'disobedience'];
+const wordsFind = [
+  "machine",
+  "subset",
+  "trouble",
+  "starting",
+  "matter",
+  "eating",
+  "truth",
+  "disobedience",
+];
+
```

Comment:
An alternative solution is ` function doesWordExist(arr, word){
  for (let el of arr){
     if (el === word) return true;
   }
  return false; }`

Reply:
Thanks for the feedback! Your suggestions made my code a lot cleaner